src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+        // Type parameter 'type parameter' has no matching typeparam tag in the XML comment on 'type' (but other type parameters do)
+        private const string CS1712 = nameof(CS1712);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS1573, CS1712);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        private BaseParameterListSyntax FindFirstAncestorParameterItem(SyntaxNode node)
+        {
+            for (; node != null; node = node.Parent)
+            {
+                if (node is BaseMethodDeclarationSyntax)
+                {
+                    return ((BaseMethodDeclarationSyntax)node).ParameterList;
+                }
+                if (node is IndexerDeclarationSyntax)
+                {
+                    return ((IndexerDeclarationSyntax)node).ParameterList;
+                }
+                if (node is DelegateDeclarationSyntax)
+                {
+                    return ((DelegateDeclarationSyntax)node).ParameterList;
+                }
+            }
+            return null;
+        }
+
+        // Can't use the generic GetAncestor method because structs and classes can nest each other.
+        private TypeParameterListSyntax FindFirstAncestorTypeParameterItem(SyntaxNode node)
+        {
+            for (; node != null; node = node.Parent)
+            {
+                if (node is MethodDeclarationSyntax)
+                {
+                    return ((MethodDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is ClassDeclarationSyntax)
+                {
+                    return ((ClassDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is StructDeclarationSyntax)
+                {
+                    return ((StructDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is InterfaceDeclarationSyntax)
+                {
+                    return ((InterfaceDeclarationSyntax)node).TypeParameterList;
+                }
+                if (node is DelegateDeclarationSyntax)
+                {
+                    return ((DelegateDeclarationSyntax)node).TypeParameterList;
+                }
+            }
+            return null;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            // TODO many CodeFixProvider seem to do that and it works. Shouldn't there be a loop over all of the fixes?
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+
+            if (diagnostic.Id == CS1573)
+            {
+                var parameterList = FindFirstAncestorParameterItem(root.FindToken(diagnosticSpan.Start).Parent);
```

Comment:
Note: you don't need custom loops to walk up ancestors. For example, you could just use `Ancestors(...)` and `AncestorsAndSelf(...)` like so:

``` C#
var parameterList = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<MemberDeclarationSyntax>().FirstOrDefault(m => m.GetParameterList());
```